src/Arcus.EventGrid.Core/Publishing/EventGridPublisherClientWithTracking.cs

```diff
@@ -0,0 +1,629 @@
+            ICorrelationInfoAccessor correlationAccessor,
+            EventGridPublisherClientWithTrackingOptions options,
+            ILogger<EventGridPublisherClient> logger)
+        {
+            Guard.NotNullOrWhitespace(topicEndpoint, nameof(topicEndpoint), "Requires a non-blank Azure Event Grid topic endpoint to register the Azure Event Grid publisher with built-in correlation tracking");
+            Guard.NotNullOrWhitespace(authenticationKeySecretName, nameof(authenticationKeySecretName), "Requires a non-blank authentication key to initiate interaction with the Azure Event Grid when registering the Azure Event Grid publisher with built-in correlation tracking");
+            Guard.NotNull(secretProvider, nameof(secretProvider), "Requires an Arcus secret store implementation to retrieve the authentication key to initiate interaction with Azure Event Grid");
+            Guard.NotNull(correlationAccessor, nameof(correlationAccessor), "Requires an correlation accessor implementation to retrieve the current correlation model when enriching the publishing event");
+            Guard.NotNull(options, nameof(options), "Requires a set of additional options to influence the correlation tracking and internal HTTP request which represents the publishing event");
+            Guard.NotNull(logger, nameof(logger), "Requires a logger instance to track the Azure Event Grid dependency with correlation information");
+
+            _topicEndpoint = topicEndpoint;
+            _authenticationKeySecretName = authenticationKeySecretName;
+            _secretProvider = secretProvider;
+        
+            CorrelationAccessor = correlationAccessor;
+            Options = options;
+            Logger = logger;
+        }
+
+        /// <summary>
+        /// Gets the available accessor instance to retrieve the current correlation information model used during the tracking of the publishing event.
+        /// </summary>
+        protected ICorrelationInfoAccessor CorrelationAccessor { get; }
+
+        /// <summary>
+        /// Gets the additional options to influence the correlation tracking and internal HTTP request which represents the publishing event.
+        /// </summary>
+        protected EventGridPublisherClientWithTrackingOptions Options { get; }
+
+        /// <summary>
+        /// Gets the logger instance to track the Azure Event Grid dependency with correlation information.
+        /// </summary>
+        protected ILogger Logger { get; }
+
+        /// <summary>
+        /// Publishes a CloudEvent to an Event Grid topic.
+        /// </summary>
+        /// <param name="cloudEvent"> The set of events to be published to Event Grid. </param>
+        /// <param name="cancellationToken"> An optional cancellation token instance to signal the request to cancel the operation.</param>
+        public override Response SendEvent(CloudEvent cloudEvent, CancellationToken cancellationToken = new CancellationToken())
```

Comment:
I guess the `new CancellationToken` specification here is because the override must exactly match the method - signature of the method that you override.  (In most cases, default value for cancellationToken is CancellationToken.Default or something ?

Reply:
Aha, yeah, we can change that to `default`. Let's do that!
Adapted in https://github.com/arcus-azure/arcus.eventgrid/pull/245/commits/4dc363d2859e93f527007d773b07541fa06e5c6f